transport/roundtrip_test.go

```diff
@@ -218,104 +220,91 @@ func TestSimpleRoundTrip(t *testing.T) {
 		wantError func(error)
 	}{
 		{
+			name:            "headers",
 			requestHeaders:  transport.NewHeaders().With("token", "1234"),
 			requestBody:     "world",
 			responseHeaders: transport.NewHeaders().With("status", "ok"),
 			responseBody:    "hello, world",
 		},
 		{
+			name:          "internal err",
 			requestBody:   "foo",
 			responseError: yarpcerrors.Newf(yarpcerrors.CodeInternal, "great sadness"),
 			wantError: func(err error) {
 				assert.True(t, yarpcerrors.FromError(err).Code() == yarpcerrors.CodeInternal, err.Error())
 			},
 		},
 		{
+			name:          "invalid arg",
 			requestBody:   "bar",
 			responseError: yarpcerrors.Newf(yarpcerrors.CodeInvalidArgument, "missing service name"),
 			wantError: func(err error) {
 				assert.True(t, yarpcerrors.FromError(err).Code() == yarpcerrors.CodeInvalidArgument, err.Error())
 			},
 		},
-		{
-			requestBody: "baz",
-			responseError: yarpcerrors.Newf(yarpcerrors.CodeInternal,
-				`error for procedure "foo" of service "bar": great sadness`,
-			),
-			wantError: func(err error) {
-				assert.True(t, yarpcerrors.FromError(err).Code() == yarpcerrors.CodeInternal, err.Error())
-			},
-		},
-		{
-			requestBody: "qux",
-			responseError: yarpcerrors.Newf(yarpcerrors.CodeInvalidArgument,
-				`BadRequest: unrecognized procedure "echo" for service "derp"`,
-			),
-			wantError: func(err error) {
-				assert.True(t, yarpcerrors.FromError(err).Code() == yarpcerrors.CodeInvalidArgument, err.Error())
-			},
-		},
 	}
 
-	rootCtx := context.Background()
 	for _, tt := range tests {
 		for _, trans := range transports {
-			requestMatcher := transporttest.NewRequestMatcher(t, &transport.Request{
-				Caller:    testCaller,
-				Service:   testService,
-				Transport: trans.Name(),
-				Procedure: testProcedure,
-				Encoding:  raw.Encoding,
-				Headers:   tt.requestHeaders,
-				Body:      bytes.NewReader([]byte(tt.requestBody)),
-			})
-
-			handler := unaryHandlerFunc(func(_ context.Context, r *transport.Request, w transport.ResponseWriter) error {
-				r.Headers.Del("user-agent") // for gRPC
-				r.Headers.Del(":authority") // for gRPC
-				assert.True(t, requestMatcher.Matches(r), "request mismatch: received %v", r)
-
-				if tt.responseError != nil {
-					return tt.responseError
-				}
-
-				if tt.responseHeaders.Len() > 0 {
-					w.AddHeaders(tt.responseHeaders)
-				}
-
-				_, err := w.Write([]byte(tt.responseBody))
-				assert.NoError(t, err, "failed to write response for %v", r)
-				return err
-			})
-
-			ctx, cancel := context.WithTimeout(rootCtx, 200*testtime.Millisecond)
-			defer cancel()
-
-			router := staticRouter{Handler: handler}
-			trans.WithRouter(router, func(o transport.UnaryOutbound) {
-				res, err := o.Call(ctx, &transport.Request{
+			t.Run(tt.name+"-"+trans.Name(), func(t *testing.T) {
```

Comment:
If you use "/" here instead of "-", the tests will be treated like regular subtests with support for matching on subtest names.